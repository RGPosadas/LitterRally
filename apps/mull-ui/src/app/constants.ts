export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
