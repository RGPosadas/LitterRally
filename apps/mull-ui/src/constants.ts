/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;
