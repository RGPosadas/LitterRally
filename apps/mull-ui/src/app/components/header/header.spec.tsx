import React from 'react';
import { render } from '@testing-library/react';

import Header from './header';
import { Router } from 'react-router-dom';

import { createMemoryHistory } from 'history';
import { ROUTES } from '../../../constants';

import renderer from 'react-test-renderer';

describe('Header', () => {
  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.DISCOVER);

    const { baseElement } = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    expect(baseElement).toBeTruthy();
  });

  it('button associated with current should be active ', () => {
    const testId = 'profile-mobile-navlink';

    const history = createMemoryHistory();
    const dom = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    history.push(ROUTES.PROFILE);
    const element = dom.getByTestId(testId);
    expect(element.classList.contains('active')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
        <Router history={history}>
          <Header />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
