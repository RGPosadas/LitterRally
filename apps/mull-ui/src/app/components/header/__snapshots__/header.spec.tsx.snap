// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div
  className="header"
>
  <img
    alt="Mull logo"
    className="top-logo nav-element"
    src="mull-logo.png"
  />
  <a
    aria-current={null}
    data-testid="profile-mobile-navlink"
    href="/profile"
    onClick={[Function]}
  >
    <svg
      className="profile-button-mobile"
    >
      ProfileIcon.svg
    </svg>
  </a>
</div>
`;
